ServerCore/ServerMessages/PresenceStore.cs

```diff
@@ -88,10 +88,52 @@ public async Task MergePresenceState(PresenceMessage[] allPresence)
                 await OnPresenceChange(message);
             }
         }
+    }
+
+    /// <summary>
+    /// Model for a user's presence on a puzzle page
+    /// </summary>
+    public class PresenceModel
+    {
+        public int UserId { get; set; }
+        public string Name { get; set; }
```

Comment:
A _little_ weird that this is here even though the PresenceStore never sets it, but in the short term I won't quibble

Reply:
Yeah, it's a bit messy. I really wish we had a good way to get the name here in the first place (we could send it over SignalR, but I'm trying to keep presence messages small since there are a ton of them)